src/liboslexec/llvm_gen.cpp

```diff
@@ -2483,9 +2481,7 @@ LLVMGEN (llvm_gen_texture3d)
     args.push_back (rop.sg_void_ptr());
     RendererServices::TextureHandle *texture_handle = NULL;
     if (Filename.is_constant() && rop.shadingsys().opt_texture_handle()) {
-        texture_handle = rop.renderer()->get_texture_handle (*(ustring *)Filename.data());
-        if (! rop.renderer()->good (texture_handle))
-            texture_handle = NULL;
+        texture_handle = rop.renderer()->get_texture_handle (*(ustring *)Filename.data(), rop.shadingcontext());
```

Comment:
Does the logic change here slightly? It no longer tests for "good()", which encompasses both if the handle is null, but also if the file is broken somehow. So handles of broken files previously got changed to null here, but not now? Is that bad?

Reply:
As you said -- the only thing the `filename` variant of the code on the OIIO side does is translate a filename to a handle before calling the handle version.

In other words -- if you already have a handle to the broken file, OIIO knows how to deal with that correctly and would not do anything differently if you called the filename version instead. So it seems like a pessimization to reset the handle to `NULL` in that case.

Reply:
OK, sounds reasonable.